### PR TITLE
Deprecate global_ordinals_hash and global_ordinals_low_cardinality

### PR DESCRIPTION
**Pull request by jimczi**
This change deprecates the `global_ordinals_hash` and `global_ordinals_low_cardinality` and
makes the `global_ordinals` execution hint choose internally if global ords should be remapped or use the segment ord directly.
These hints are too sensitive and expert to be exposed and we should be able to take the right decision internally based on the agg tree.

Relates #26014

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java [hunk 1]
```diff
@@ -291,44 +289,35 @@ Aggregator create(String name,
...
-                    return mode;
-                }
+        public static ExecutionMode fromString(String value, final DeprecationLogger deprecationLogger) {
+            if ("global_ordinals".equals(value)) {
+                return GLOBAL_ORDINALS;
+            } else if ("global_ordinals_hash".equals(value)) {
+                deprecationLogger.deprecated("global_ordinals_hash is deprecated. Please use [global_ordinals] instead.");
+                return GLOBAL_ORDINALS;
+            } else if ("map".equals(value)) {
+                return MAP;
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java]: let's do a switch/case?

**Review by jpountz — COMMENTED** [review]
I have mixed feelings about this PR. On the one hand I like it because I'd like users to never have to care about the execution mode, but on the other hand this PR moves part of the decision process from the factory to the aggregator, which feels wrong to me, and reduces testing coverage since we cannot force some execution modes anymore.

**Comment by jimczi** [comment]
> but on the other hand this PR moves part of the decision process from the factory to the aggregator, which feels wrong to me, and reduces testing coverage since we cannot force some execution modes anymore.

isn't it the opposite ? The factory now picks global ordinals execution with remapping or plain ordinals with the same heuristic than before. I removed the late decision in the aggregator and replaced it with https://github.com/elastic/elasticsearch/pull/26014. The aggregator is just able to remap or not based on the decision made in the factory.
For testing coverage I can add tests for the `GlobalOrdinalsStringTermsAggregator` which lacks testing anyway ?

**Comment by jpountz** [comment]
Woops, I got a bit lost in the diff and misread your PR, sorry. This change is actually great the way it is. Thanks!

+1 to add tests to `GlobalOrdinalsStringTermsAggregator`.

**Pull request merged**